src/DGtal/geometry/curves/Naive3DDSSComputer.ih

```diff
@@ -0,0 +1,587 @@
+    {
+        Point2d tyz = myYZalgo.front();
+        myYZalgo.extendFront();
+        if ( myYZalgo.front() != tyz )
+        {
+            test++;
+            lenYZ++;
+        }
+        else
+	{
+	    myYZalgo.retractFront();
+            blockYZ = true;
+	}
+    }
+    if ( test >= 2 )
+    {
+        myEnd++;
+        return true;
+    }
+    else return false;
+}
+
+/** Tests whether the 3D DSS can be extended at the front.
+ *
+ * @return 'true' if yes, 'false' otherwise
+ */
+template <typename TIterator, typename TInteger, int connectivity>
+inline
+bool
+DGtal::Naive3DDSSComputer<TIterator, TInteger,connectivity>::isExtendableFront()
+{
+    //projection on xy-plane
+    short test = 0;
+    if ( myXYalgo.isExtendableFront() && !blockXY )
+        test++;
+    //projection on xz-plane
+    if ( myXZalgo.isExtendableFront() && !blockXZ )
+        test++;
+    //projection on yz-plane
+    if ( myYZalgo.isExtendableFront() && !blockYZ )
+        test++;
+    if ( test >= 2 )
+        return true;
+    else return false;
+}
+
+template <typename TIterator, typename TInteger, int connectivity>
+inline
+TIterator
+DGtal::Naive3DDSSComputer<TIterator,TInteger,connectivity>::begin() const {
+  return myBegin;
+}
+
+template <typename TIterator, typename TInteger, int connectivity>
+inline
+TIterator
+DGtal::Naive3DDSSComputer<TIterator,TInteger,connectivity>::end() const {
+  return myEnd;
+}
+
+//-----------------------------------------------------------------
+/**
+ * Checks the validity/consistency of the object.
+ * @return 'true' if the object is valid, 'false' otherwise.
+ */
+
+template <typename TIterator, typename TInteger, int connectivity>
+inline
+bool
+DGtal::Naive3DDSSComputer<TIterator,TInteger,connectivity>::isValid() const
+{
+  return ( ( myXYalgo.isValid() ) &&
+           ( myXZalgo.isValid() ) &&
+           ( myYZalgo.isValid() ) );
+}
+
+/**
+ * Computes the parameters
+ * (direction, intercept, thickness)
+ * of the DSS
+ * @param direction
+ * @param intercept
+ * @param thickness
+ */
+template <typename TIterator, typename TInteger, int connectivity>
+inline
+void
+DGtal::Naive3DDSSComputer<TIterator,TInteger,connectivity>
+::getParameters ( Point3d& direction, PointD3d& intercept, PointD3d& thickness ) const
+{
+    if ( lenXY > lenYZ && lenXZ > lenYZ ) { //XY-plane, XZ-plane
+
+        Integer a1 = myXYalgo.b();
+        Integer b1 = myXYalgo.a();
+        Integer a2 = myXZalgo.b();
+        Integer c1 = myXZalgo.a();
+
+        if ( c1 == 0 || ( a1 == 0 && a2 == 0 ) )
+            direction = Point3d ( a1, b1, c1 );
+        else
+        {
+            if ( b1 == 0  )
+                direction = Point3d ( a2, b1, c1 );
+            else
+            {
+                direction = Point3d ( a1 * a2 , a2 * b1 , a1 * c1 );
+            }
+        }
+
+        Integer mu1 = myXYalgo.mu();
+        Integer mu2 = myXZalgo.mu();
+        double y = (double) -NumberTraits<TInteger>::castToInt64_t(mu1) / a1;
```

Comment:
Suggest () around numerator and ((double) ...) around denominator so that it is obvious that you want a floating-point division.